w06-ruby-on-rails/d3-dawn-ruby-oop/README.md

```diff
@@ -0,0 +1,176 @@
+What if I wanted a running counter for all the cars I've ever created? Enable the code: `Car.total` which outputs the total number of all cars ever created.
+
+*Hint: Create a class variable with `@@` to store the count and use the `self` keyword to add a class method*
+
+*Note: Class variables are used much more infrequently than instance variables*
+
+## Quick Review (15m)
+
+Think, pair, share...
+
+  * What is a class?
+    - What is an attribute?
+    - What is a method?
+  * What is the difference between:
+    - an instance variable,
+    - a class variable
+  * Why do we use classes?
+  * What is inheritance?
+
+## Inheritance (25m)
+
+Given an `Vehicle` class that contains the method `accelerate` & attribute `speed`..
+
+```ruby
+class Vehicle
+  attr_accessor :speed
+  def initialize
+  	@speed = 0
+  end
+  def accelerate(speed)
+  	@speed += speed
+  end
+end
+```
+
+**Challenge:**
+How can I create both `Car` and `Motorbike` classes while being DRY and not duplicating the method `accelerate` in each?
+
+## Exercise: Characters and Superheroes
+
+Previously you worked with using inheritance to create some `Animal` and `People` classes. This exercise will review those concepts, using the `Character` and `Superhero` classes.
+
+Superheroes are still Characters after all. In this exercise, you'll define:
+
+  1. An `Character` class, with the following:
+    * Properties:
+      * `name`: A string that holds the character's name
+    * Instance Methods:
+      * `location`: Takes a parameter `origin` prints out a message telling the character's `name` and where they are from (their `origin`)
+      * `exercise` & `rest`: These two methods should NOT be passed any arguments. Instead, they will set an instance variable `@state` to the string `"ready"` or `"tired"` respectively.
+
+  2. A `Superhero` class, with the following characteristics:
+    * Inherits from `Character`
+    * Automatically sets `@alignment` to `"good"`
+    * Adds 3 new instance vars:
+      * species
+      * charisma
+      * secret_identity
+    - Superheroes must keep their identities secret! Make sure your `Superhero` class *overrides* the existing `greet` method (in `Character`) so that a `Superhero` doesn't reveal their true `"name"`
+
+**BONUS:**
+
+* Sometimes superheroes must reveal who they are to the public. Add an instance method called `reveal` that prints out a superhero's name and the secret identity they used in the following format: "I am Green Arrow. But you know me better as Oliver Queen." (Hint: look up how to interpolate strings in Ruby)
+ * Add a `class variable` called `count` that keeps track of all the superheroes you create.
+
+ **Stretch** (no solution)
+
+ Create a `Fruit` class that has an attribute of `kind`, a string, and `diameter`, a random integer between 1 and 5.
+
+ Create a `Tree` class that has the attribute `fruits`, which is an array, the attribute `age`, which starts at 0, and the attribute `dead`, which should be a boolean.
```

Comment:
I like that you're adding one class as an array of instances on another.  I think that starts to introduce a concept that is really blurry right now.